tests/Extensions.Hosting.AsyncInitialization.Tests/AsyncInitializationAndRunTests.cs

```diff
@@ -176,50 +231,113 @@ public async Task Host_is_disposed_after_failing_initializer(bool forceIDisposab
         {
+            var host = CreateHost(services =>
+            {
+                services.AddAsyncInitializer(sp => A.Fake<IAsyncTeardown>());
+                services.AddHostedService<TestService>();
+            }, forceIDisposableHost: forceIDisposableHost);
 
+            OutputHelper.WriteLine(host is IAsyncDisposable ? "Using IAsyncDisposable Host" : "Using IDisposable Host");
+
+            var lifetime = host.Services.GetRequiredService<IHostApplicationLifetime>();
             using var cancellationTokenSource = new CancellationTokenSource();
-            var service = A.Fake<TestService>();
-            A.CallTo(() => service.StartAsync(A<CancellationToken>._)).Invokes(_ => cancellationTokenSource.Cancel()).CallsBaseMethod();
+            using var ctr = lifetime.ApplicationStarted.Register(cancellationTokenSource.Cancel);
+
+            await host.InitAndRunAsync(cancellationTokenSource.Token);
 
+            Assert.Throws<ObjectDisposedException>(host.Services.CreateScope);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task Host_is_disposed_after_service_fails(bool forceIDisposableHost)
+        {
             var host = CreateHost(services =>
             {
                 services.AddAsyncInitializer(sp => A.Fake<IAsyncTeardown>());
-                services.AddHostedService(factory => service);
+                services.AddHostedService<FaultingService>();   
             }, forceIDisposableHost: forceIDisposableHost);
 
             OutputHelper.WriteLine(host is IAsyncDisposable ? "Using IAsyncDisposable Host" : "Using IDisposable Host");
 
-            await Assert.ThrowsAsync<TaskCanceledException>(() => host.InitAndRunAsync(cancellationTokenSource.Token));
-
-            var exception = Record.Exception(() => host.Services.CreateScope());
-            Assert.IsType<ObjectDisposedException>(exception);
+            await Assert.ThrowsAsync<ApplicationException>(() => host.InitAndRunAsync()); 
+            Assert.Throws<ObjectDisposedException>(host.Services.CreateScope);  
         }
 
+
         [Fact]
         public async Task Initializer_with_teardown_and_scoped_dependency_is_resolved()
         {
+
             var host = CreateHost(
                 services =>
                 {
-                    services.AddScoped<IDependency, DisposableDependency>();
+                    services.AddScoped(sp => A.Fake<IDependency>());
                     services.AddAsyncInitializer<InitializerWithTearDown>();
-                    services.AddTransient(factory => OutputHelper);
-                    services.AddHostedService<TestService>();
+                    services.AddHostedService<StoppingService>();
                 },
                 true);
 
             await host.InitAndRunAsync();    
         }
 
+        [Fact]
+        public async Task Scoped_disposable_dependency_is_disposed_twice_before_service()
```

Comment:
Not sure what the purpose of this test is. Disposing services is done by the DI container, not this library. Why do you need to test it?
Also, why should it be disposed twice?

Reply:
Admittingly, this is not really a functional unit test. It evolved during our discussion around scoping and more or less proves that objects are disposed of in the correct order depending on their lieftime.  It's counterpart is `Singleton_disposable_dependency_is_disposed_once_after_service()`
Both tests could actually be removed.

Reply:
I decided to flag both tests as Obsolete() and made them private.

Reply:
Why not just remove them, then?